frameworks/geniverse/resources/animation_core.js

```diff
@@ -2144,17 +2142,29 @@ sc_require('lib/burst-core');
     // Setup Draw-Loop
     ////////////////////////////////////////////////////////////////////////////
 
+    var $this = $(this);
+
     // cf. http://creativejs.com/resources/requestanimationframe/
     function drawLoop() {
-      timeoutID = setTimeout(function() {
-        requestID = requestAnimationFrame(drawLoop);
+      // use $.data() to store animation IDs with DOM node
+      $this.data('timeoutID', setTimeout(function() {
+        $this.data('requestID', requestAnimationFrame(drawLoop));
         // Drawing code goes here
         draw();
-      }, 1000 / fps);
+      }, 1000 / fps));
+    }
+
+    // cancel any previous animations
+    if ($this.data('requestID')) {
+      cancelAnimationFrame($this.data('requestID'));
+      $this.data('requestID', null);
     }
-    // don't start a timer if we've already got one
-    if (!timeoutID)
```

Comment:
Would you mind confirming my understanding here? Previously, the giant function in this file was called each time the play or reset button was hit. It started a new animation request loop, which was never cancelled, even when a new one started. In the new code, the window's data remembers the ID's for animations that start, so that when a new one begins, it can first cancel the old ones.

Is that correct? If so, did the condition I'm commenting on ever fail? Or was timeoutID always going to be uninitialized in this previous code.